finagle-mysql/src/main/scala/com/twitter/finagle/mysql/Row.scala

```diff
@@ -130,14 +143,19 @@ class BinaryEncodedRow(rawRow: Buf, val fields: IndexedSeq[Field], indexMap: Map
     for ((field, idx) <- fields.zipWithIndex) yield {
       if (isNull(idx)) NullValue
       else field.fieldType match {
-        case Type.Tiny        => ByteValue(reader.readByte())
-        case Type.Short       => ShortValue(reader.readShortLE())
-        case Type.Int24       => IntValue(reader.readIntLE())
-        case Type.Long        => IntValue(reader.readIntLE())
-        case Type.LongLong    => LongValue(reader.readLongLE())
-        case Type.Float       => FloatValue(reader.readFloatLE())
-        case Type.Double      => DoubleValue(reader.readDoubleLE())
-        case Type.Year        => ShortValue(reader.readShortLE())
+        case Type.Tiny if isSigned(field)     => ByteValue(reader.readByte())
+        case Type.Tiny                        => ShortValue(reader.readUnsignedByte())
+        case Type.Short if isSigned(field)    => ShortValue(reader.readShortLE())
+        case Type.Short                       => IntValue(reader.readUnsignedShortLE())
+        case Type.Int24 if isSigned(field)    => IntValue(reader.readIntLE())
```

Comment:
`Type.Int24` and `TypeInt24 if signed need to call `reader.readMediumLE()` and `reader.readUnsignedMediumLE()`, respectively.